extensions/amp-user-notification/0.1/amp-user-notification.js

```
@@ -0,0 +1,423 @@
+      method: 'POST',
+      credentials: 'include',
+      body: {
+        'elementId': this.elementId_,
+        'ampUserId': this.ampUserId_
+      }
+    });
+  }
+
+  /**
+   * Success handler for `postShowEndpoint_`.
+   * @param {!PostResponseMetadata}
+   * @return {!Promise<boolean>}
+   * @private
+   */
+  onPostShowEndpointSuccess_(data) {
+    assert(typeof data['showNotification'] == 'boolean', '`showNotification` ' +
+        'should be a boolean. Got "%s" which is of type %s.',
+        data['showNotification'], typeof data['showNotification']);
+
+    if (data['showNotification']) {
+      return data['showNotification'];
+    }
+    // if no notification is shown, resolve right away with `false` since
+    // we dont need to show it.
+    return this.dialogResolve_(false);
+  }
+
+  /**
+   * Generic rejection handler for amp-user-notification.
+   * @return {boolean}
+   * @private
+   */
+  onRejection_(msg) {
+    log.error(TAG_, 'An error ocurred on `amp-user-notification` ' +
+        `with html element id "${this.elementId_}". message: ${msg}`);
+    return false;
+  }
+
+  /**
+   * Get async cid service.
+   * @return {!Promise}
+   * @private
+   */
+  getAsyncCid_() {
+    return cidFor(this.win_).then(cid => {
+      // `amp-user-notification` is our cid scope, while we give it a resolved
+      // promise for the 2nd argument so that the 3rd argument (the
+      // persistentConsent) is the one used to resolve getting
+      // the external CID.
+      return cid.get('amp-user-notification',
+          Promise.resolve(), this.dialogPromise_);
+    });
+  }
+
+  /**
+   * @return {!Promise<boolean>}
+   * @private
+   **/
+  getIfHref_() {
+    if (!this.getIfHrefPromise_) {
+      this.getIfHrefPromise_ = this.getAsyncCid_()
+          .then(this.postShowEndpoint_.bind(this))
+          .then(this.onPostShowEndpointSuccess_.bind(this))
+          .catch(this.onRejection_.bind(this));
```

Should we just let it fail?